src/main/java/uk/gov/hmcts/reform/ccd/document/am/controller/endpoints/CaseDocumentAmController.java

```diff
@@ -230,28 +204,21 @@ public CaseDocumentAmController(DocumentManagementService documentManagementServ
     public ResponseEntity<Object> generateHashCode(
 
         @PathVariable("documentId") UUID documentId) {
-        try {
-            StoredDocumentHalResource resource = new StoredDocumentHalResource();
-            ResponseEntity responseEntity = documentManagementService.getDocumentMetadata(documentId);
-            if (responseEntity.getStatusCode().equals(HttpStatus.OK) && null != responseEntity.getBody()) {
-                resource = (StoredDocumentHalResource) responseEntity.getBody();
-            }
-
-            ValidationService.validateInputParams(INPUT_STRING_PATTERN, documentId.toString(),
-                                                  resource.getMetadata().get("caseTypeId"), resource.getMetadata().get("jurisdictionId"));
+        StoredDocumentHalResource resource = new StoredDocumentHalResource();
+        ResponseEntity responseEntity = documentManagementService.getDocumentMetadata(documentId);
+        if (responseEntity.getStatusCode().equals(HttpStatus.OK) && null != responseEntity.getBody()) {
+            resource = (StoredDocumentHalResource) responseEntity.getBody();
+        }
 
-            HashMap<String, String> responseBody = new HashMap<>();
+        ValidationService.validateInputParams(INPUT_STRING_PATTERN, documentId.toString(),
```

Comment:
Please prepare string as below to generate hash token as it's using in more than one place inside method and use it.
StringBuilder sb = new  StringBuilder();
sb.append(documentId.toString().concat(
             resource.getMetadata().get("jurisdictionId")).concat(resource.getMetadata().get("caseTypeId")));

Reply:
This is actually being used only once while generating hash token. In this instance we are simply passing the individual strings like caseTypeId, JID and document Id for the validation.